pkg/dynamicconfig/reflectors.go

```diff
@@ -66,3 +156,64 @@ func configMapReflector(name string, cs k8sClient.Clientset, t statedb.RWTable[D
 		},
 	}
 }
+
+func ciliumNodeReflector(name string, cs k8sClient.Clientset, t statedb.RWTable[DynamicConfig], l *slog.Logger, index int, sourceLen int, overrides resolver.ConfigOverride) k8s.ReflectorConfig[DynamicConfig] {
+	return k8s.ReflectorConfig[DynamicConfig]{
+		Name:  "node-" + name,
+		Table: t,
+		TransformMany: func(o any) []DynamicConfig {
+			var entries []DynamicConfig
+			node := o.(*corev1.Node).DeepCopy()
+
+			for k, v := range parseNodeConfig(node, l) {
+				priority := getPriorityForKey(k, overrides, index, sourceLen)
+				dc := DynamicConfig{Key: Key{Name: k, Source: node.Name}, Value: v, Priority: priority}
+				entries = append(entries, dc)
+			}
+			return entries
+		},
+		ListerWatcher: utils.ListerWatcherWithModifiers(
+			utils.ListerWatcherFromTyped[*corev1.NodeList](cs.Slim().CoreV1().Nodes()),
+			func(opts *metav1.ListOptions) {
+				opts.FieldSelector = fields.ParseSelectorOrDie(metadataName + name).String()
+			},
+		),
+		QueryAll: func(txn statedb.ReadTxn, t statedb.Table[DynamicConfig]) iter.Seq2[DynamicConfig, statedb.Revision] {
+			return statedb.Filter(
+				t.All(txn),
+				func(dc DynamicConfig) bool {
+					return dc.Key.Source == name
+				},
+			)
+		},
+	}
+}
+
+// parseNodeConfig returns a map of overridable fields from the Node object
+// It allows annotation or labels with config.cilium.io/K=V
+func parseNodeConfig(node *corev1.Node, logger *slog.Logger) map[string]string {
+	out := map[string]string{}
+	read := func(in map[string]string) {
+		for k, v := range in {
+			if !strings.HasPrefix(k, annotation.ConfigPrefix) {
+				continue
+			}
+
+			s := strings.SplitN(k, "/", 2)
+			if len(s) != 2 {
+				logger.Warn("node annotation format invalid", logfields.ConfigAnnotation, k)
```

Comment:
I think it would be useful to provide the expected format here so that the user can rectify any mistakes themselves.

```suggestion
				logger.Warn("Detected invalid format in annotation, expected <INSERT EXPECTED FORMAT>. Skipping key", logfields.ConfigAnnotation, k)
```